### PR TITLE
[Swift] Use Environment Variables for Vapor

### PR DESCRIPTION
**Pull request by 0xTim**
Replaces #2830 

edit
--------
Closes #2830

### CODE REVIEW

File: swift/vapor-framework/Package.resolved [hunk 1]
```diff
@@ -114,8 +114,8 @@
         "repositoryURL": "https://github.com/vapor/vapor.git",
```
Comment (waghanza) [inline swift/vapor-framework/Package.resolved:114]: could you remove this file ?

File: swift/vapor-framework/Sources/server/main.swift [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import Vapor
 
-var env = Environment.production
+var env = Environment(name: Environment.get("VAPOR_ENV") ?? "production")
```
Comment (waghanza) [inline swift/vapor-framework/Sources/server/main.swift:3]: should not it be **dev** by default ? I mean the environment variable will always be set `here`, but a common usage is to set to **dev** environment (at least in `ruby`)

File: swift/vapor-framework/Package.resolved (hunk 1 above)
Comment (0xTim) [reply swift/vapor-framework/Package.resolved]: Done!

File: swift/vapor-framework/Sources/server/main.swift (hunk 2 above)
Comment (0xTim) [reply swift/vapor-framework/Sources/server/main.swift]: Ah fair enough, done as well

**Comment by waghanza** [comment]
last thing, could you remove **trailing space** in `config.yaml`
https://www.codefactor.io/repository/github/the-benchmarker/web-frameworks/pull/2919

**Comment by 0xTim** [comment]
@waghanza think that's done!

**Comment by waghanza** [comment]
still missing one https://www.codefactor.io/repository/github/the-benchmarker/web-frameworks/pull/2919

**Pull request merged**